lib/datasets/japanese-date-parser.rb

```diff
@@ -0,0 +1,42 @@
+module Datasets
+  class JapaneseDateParser
+    class UnsupportedEraInitialRange < StandardError; end
```

Comment:
Could you use `Datasets::Error` not `StandardError` as the parent class?

```suggestion
    class UnsupportedEraInitialRange < Error; end
```